### DISCUSSION TITLE
using abacus to do COHP calculation

### DISCUSSION THREAD

**Discussion by Satinelamp** [deepmodeling/abacus-develop#1598] [General]
### Background

Crystal Orbital Hamilton Populations (COHP) analysis is very useful to see the chemical bond in the periodic system, currently, we often use `VASP` and `Lobster` (http://www.cohp.de/) to calculate COHP. Basically, VASP could generate a file called `PROOUT` which contains the necessary information to calculate COHP, Lobster program could read and calculate COHP according to the data stored in `PROOUT`.
I haven't seen a similar workflow that could use ABACUS to calculate COHP. 

### Describe the solution you'd like

There are two different ways to implement this function in ABACUS:
1. Abacus could generate the same `PROOUT` file (https://www.vasp.at/wiki/index.php/PROOUT) and use the same data format as VASP (https://www.cnblogs.com/panscience/p/13332217.html), then we could use `Lobster` to do the rest.
2.  Abacus could generate similar data as VASP, but with a different data format, which requires a new post-processing tool.

### Additional Context

This should be included in the development of the post-processing tool of ABACUS, but currently, it seems there aren't any GitHub repositories for the post-processing tools of ABACUS.

**Comment by wenfei-li** [comment]
We welcome your contribution if you want to implement this feature. However, we have our own schedule of development and unfortunately COHP will not have any priority on our list.

**Comment by Satinelamp** [comment]
Thanks a lot for the response. I get it the development team is focusing on AI-assisted electronic structure calculation, which is quite a fantastic area and interesting to explore. But the usual toolset for post-processing is also necessary for newbies to migrate from other software like VASP to ABACUS. I am waiting for Prof. He, Lixin from USTC to release his post-processing toolset, so maybe we could add this feature into his toolset's framework.
